### PR TITLE
add transform-stats operation type

### PR DESCRIPTION
**Pull request by hendrikmuhs**
add a transform-stats operation type with a optional condition. The condition can
be used to wait until transform has processed all data.

### CODE REVIEW

File: esrally/driver/runner.py [hunk 1]
```diff
@@ -2116,6 +2117,58 @@ def __repr__(self, *args, **kwargs):
...
+                "unit": "ops",
+                "condition": {
+                    "path": path,
+                    # avoid mapping issues in the ES metrics store by always rendering values as strings
+                    "actual-value": self._safe_string(actual_value),
+                    "expected-value": self._safe_string(expected_value)
+                },
+                # currently we only support "==" as a predicate but that might change in the future
+                "success": actual_value == expected_value,
+                "transform-stats": transform_stats
```
Comment (b-deam) [inline esrally/driver/runner.py:2158]: Is it always required to have the transform stats returned? The main concern here is around key/field explosion on the metrics store. 

Would it make sense to either not return it at all, or perhaps could it be returned only if a user passes in a specific parameter? e.g. the `BulkIndex` runner has a `detailed-results` param to determine whether or not we return detailed stats:
https://github.com/elastic/rally/blob/40334f18f92e82686c7a1b4440d53395767d2445/esrally/driver/runner.py#L462-L467

**Review by b-deam — COMMENTED** [review]
Thanks for the PR @hendrikmuhs! I have a couple of comments, but it looks great.

File: esrally/driver/runner.py (hunk 1 above)
Comment (hendrikmuhs) [reply esrally/driver/runner.py]: its not needed and I wasn't aware that this is pushed to the metric store. I removed it.

File: docs/track.rst [hunk 2]
```diff
@@ -2326,6 +2326,41 @@ Meta-data
...
+
+This operation is :ref:`retryable <track_operations>`.
+
+Meta-data
+"""""""""
+
+* ``weight``: Always 1.
+* ``unit``: Always "ops".
+* ``success``: A boolean indicating whether the operation has succeeded.
+* ``transform-stats``: The stats for the given transform id.
```
Comment (b-deam) [inline docs/track.rst:2362]: As we no longer return the `transform-stats` meta-data, we should remove the docs reference.

**Review by b-deam — COMMENTED** [review]
Thanks @hendrikmuhs - just a quick docs fix needed and I think we're good to go.

**Comment by b-deam** [comment]
LGTM! Thanks for iterating @hendrikmuhs :)

**Review by b-deam — APPROVED** [review]

**Pull request merged**